ta/pkcs11/src/pkcs11_token.h

```diff
@@ -128,14 +128,21 @@ enum pkcs11_proc_state {
 	PKCS11_SESSION_BUSY,
 };
 
+enum pkcs11_proc_op {
+	PKCS11_OP_INIT = 0,
+	PKCS11_OP_UPDATE,
+	PKCS11_OP_FINAL,
+	PKCS11_OP_FINAL_ONESHOT,
+};
+
 /*
  * Context of the active processing in the session
  *
  * @state - ongoing active processing function or ready state
  * @mecha_type - mechanism type of the active processing
  * @always_authen - true if user need to login before each use
  * @relogged - true once client logged since last operation update
- * @updated - true once an active operation is updated
+ * @op_state - last active operation state - update, final or one-shot
```

Comment:
I would replace `state` with `step` to prevent confusion (`PKCS11_FUNC_STEP_*`).
Operation state would rather refers the the state of the cyrpto context as per pkcs11 `C_{Get|Set}OperationState()`. What do you think?

Reply:
I think what you say makes sense. I will change this to step.

Reply:
Thinking more on it , we don't require the new enum pkcs11_proc_op at all. We can just save the last active operation and check on that. I will make the change and submit the PR fo review.